src/object/mod.rs

```diff
@@ -2,70 +2,191 @@
+        }
+
+        unsafe fn set_from(self, out: &mut bool, obj: raw::Local, val: raw::Local) -> bool {
+            neon_runtime::object::set_index(out, obj, self, val)
+        }
     }
-}
 
-impl<'a, K: Value> PropertyKey for Handle<'a, K> {
-    unsafe fn get_from(self, out: &mut raw::Local, obj: raw::Local) -> bool {
-        neon_runtime::object::get(out, obj, self.to_raw())
+    impl<'a, K: Value> PropertyKey for Handle<'a, K> {
+        unsafe fn get_from(self, out: &mut raw::Local, obj: raw::Local) -> bool {
+            neon_runtime::object::get(out, obj, self.to_raw())
+        }
+
+        unsafe fn set_from(self, out: &mut bool, obj: raw::Local, val: raw::Local) -> bool {
+            neon_runtime::object::set(out, obj, self.to_raw(), val)
+        }
+    }
+
+    impl<'a> PropertyKey for &'a str {
+        unsafe fn get_from(self, out: &mut raw::Local, obj: raw::Local) -> bool {
+            let (ptr, len) = Utf8::from(self).into_small_unwrap().lower();
+            neon_runtime::object::get_string(out, obj, ptr, len)
+        }
+
+        unsafe fn set_from(self, out: &mut bool, obj: raw::Local, val: raw::Local) -> bool {
+            let (ptr, len) = Utf8::from(self).into_small_unwrap().lower();
+            neon_runtime::object::set_string(out, obj, ptr, len, val)
+        }
     }
 
-    unsafe fn set_from(self, out: &mut bool, obj: raw::Local, val: raw::Local) -> bool {
-        neon_runtime::object::set(out, obj, self.to_raw(), val)
+    /// The trait of all object types.
+    pub trait Object: Value {
+        fn get<'a, C: Context<'a>, K: PropertyKey>(self, _: &mut C, key: K) -> NeonResult<Handle<'a, JsValue>> {
+            build(|out| { unsafe { key.get_from(out, self.to_raw()) } })
+        }
+
+        fn get_own_property_names<'a, C: Context<'a>>(self, _: &mut C) -> JsResult<'a, JsArray> {
+            build(|out| { unsafe { neon_runtime::object::get_own_property_names(out, self.to_raw()) } })
+        }
+
+        fn set<'a, C: Context<'a>, K: PropertyKey, W: Value>(self, _: &mut C, key: K, val: Handle<W>) -> NeonResult<bool> {
+            let mut result = false;
+            if unsafe { key.set_from(&mut result, self.to_raw(), val.to_raw()) } {
+                Ok(result)
+            } else {
+                Err(Throw)
+            }
+        }
     }
 }
 
-impl<'a> PropertyKey for &'a str {
-    unsafe fn get_from(self, out: &mut raw::Local, obj: raw::Local) -> bool {
-        let (ptr, len) = Utf8::from(self).into_small_unwrap().lower();
-        neon_runtime::object::get_string(out, obj, ptr, len)
+#[cfg(feature = "napi-runtime")]
+mod internal {
+    use neon_runtime::raw;
+    use handle::Handle;
+    use types::{Value, JsValue, JsArray, build};
+    use types::utf8::Utf8;
+    use context::Context;
+    use result::{NeonResult, JsResult, Throw};
+
+    /// A property key in a JavaScript object.
+    pub trait PropertyKey {
+        unsafe fn get_from<'c, C: Context<'c>>(
```

Comment:
The breaking change is these now require context. I'm not currently collecting a list of breaking changes anywhere. Do you think we should track this in a mardown doc (maybe in the repo) so that we don't need to pour over code to find them all when `n-api` is released?